CMakeLists.txt

```diff
@@ -65,62 +57,120 @@ elseif (${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
   set (CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -O3 -g -debug all")
 endif()
 
-# ----------------------------------------------------------------- #
-# --                  Compilation Instructions                   -- #
-# ----------------------------------------------------------------- #
-# -- Included files
-include_directories (${CMAKE_CURRENT_SOURCE_DIR}/include)
-
-# -- Source files
-aux_source_directory (${CMAKE_CURRENT_SOURCE_DIR}/src SRC_LIST)
-
-# -- Build a shared library
-add_library(${PROJECT_NAME} SHARED
-  ${SRC_LIST}
-  "${CMAKE_CURRENT_SOURCE_DIR}/src/amos_iso_c_fortran_wrapper.f90"
-  "${CMAKE_CURRENT_SOURCE_DIR}/src/zbesh.for"
-  "${CMAKE_CURRENT_SOURCE_DIR}/src/machine.for")
-
-# Define the properties of the library to
-# be built (version and other stuff).
-set_target_properties (
-  ${PROJECT_NAME}
-  PROPERTIES
-    VERSION   ${complex_bessel_VERSION_MAJOR}.${complex_bessel_VERSION_MINOR}.${complex_bessel_VERSION_RELEASE}
-    SOVERSION ${complex_bessel_VERSION_MAJOR}.${complex_bessel_VERSION_MINOR}.${complex_bessel_VERSION_RELEASE}
-)
-
-target_link_libraries (${PROJECT_NAME} ${LIBS})
-install (TARGETS ${PROJECT_NAME} DESTINATION lib)
-
-if (BUILD_STATIC_LIBS)
-  add_library(${PROJECT_NAME}_static STATIC
-    ${SRC_LIST}
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/amos_iso_c_fortran_wrapper.f90"
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/zbesh.for"
-    "${CMAKE_CURRENT_SOURCE_DIR}/src/machine.for")
-
-  set_target_properties(
-    ${PROJECT_NAME}_static
-    PROPERTIES
-      VERSION ${complex_bessel_VERSION_MAJOR}.${complex_bessel_VERSION_MINOR}.${complex_bessel_VERSION_RELEASE}
-  )
-
-  # Link dependencies
-  target_link_libraries (${PROJECT_NAME}_static ${LIBS})
-
-  install (TARGETS ${PROJECT_NAME}_static DESTINATION lib)
+include(GNUInstallDirs)
+SET(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}")
+SET(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}")
+SET(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}")
+SET(CMAKE_Fortran_MODULE_DIRECTORY "${PROJECT_BINARY_DIR}/modules")
+
+###############################################################################
+set(LIB complex_bessel)
+add_library(${LIB}
```

Comment:
I would prefer to build a shared library by default. Just adding `SHARED` will do, unless you want to set up an optionally `BUILD_STATIC` variable, like I had started doing but never finished...

Reply:
Done.